waku/standards/core/17/rln-relay.md

```diff
@@ -12,113 +12,150 @@ contributors:
+This means a message can only be sent every period, where the `period` is up to the application.
+
+See section [Recommended System Parameters](#recommended-system-parameters) for the RECOMMENDED method to set a sensible `period` value depending on the application.
 Peers subscribed to a spam-protected `pubsubTopic` are only allowed to send one message per `epoch`.
-The higher-level layers adopting `17/WAKU2-RLN-RELAY` MAY choose to enforce the messaging rate for  `WakuMessages` with a specific `contentTopic` published on a `pubsubTopic`.
+The higher-level layers adopting `17/WAKU2-RLN-RELAY` MAY choose to enforce the messaging rate for `WakuMessages` with a specific `contentTopic` published on a `pubsubTopic`.
 
+#### Setup and Registration
 
+A `pubsubTopic` that is spam-protected requires subscribed peers to form a [RLN group](../../../../vac/32/rln-v1.md).
+- Peers MUST be registered to the RLN group to be able to publish messages.
+- Registration MAY be moderated through a smart contract deployed on the Ethereum blockchain. 
 
-### Setup and Registration
-Peers subscribed to a specific `pubsubTopic` form a [RLN group](../../../../vac/32/rln-v1.md).
-<!-- link to the RLN group definition in the RLN RFC -->
-Peers MUST be registered to the RLN group to be able to publish messages.
-Registration is moderated through a smart contract deployed on the Ethereum blockchain. 
 Each peer has an [RLN key pair](../../../../vac/32/rln-v1.md) denoted by `sk`  and `pk`.
-The  secret key `sk` is secret data and MUST be persisted securely by the peer.
-The state of the membership contract contains the list of registered members' public identity keys i.e., `pk`s. 
-For the registration, a peer creates a transaction that invokes the registration function of the contract via which registers its `pk` in the group. 
-The transaction also transfers some amount of ether to the contract to be staked. 
+- The secret key `sk` is secret data and MUST be persisted securely by the peer.
+- The state of the membership contract SHOULD contain a list of all registered members' public identity keys i.e.,
+`pk`s. 
+
+For registration, a peer MUST create a transaction to invoke the registration function on the contract, 
+which registers its `pk` in the RLN group. 
+- The transaction MUST transfer additional tokens to the contract to be staked. 
 This amount is denoted by `staked_fund` and is a system parameter.
-The peer who has the secret key `sk` associated with a registered `pk` would be able to withdraw a portion `reward_portion` of the staked fund by providing valid proof. <!-- a secure way to prove the possession of a pk is yet under discussion, maybe via commit and reveal -->
+The peer who has the secret key `sk` associated with a registered `pk` would be able to withdraw a portion `reward_portion` of the staked fund by providing valid proof.
+<!-- a secure way to prove the possession of a pk is yet under discussion, maybe via commit and reveal -->
```

Comment:
Is this still under discussion?

Reply:
@jm-clius Slashing is not implemented in the rln-relay contract, correct? Does this mean off-chain slashing does not occur with spam detection?

Reply:
Indeed. When spam is detected, the gossipsub score of the spam-forwarding peer gets lowered. This eventually means that nodes would disconnect from peers that do not drop spam before relaying it to the network.